test/libsolidity/Metadata.cpp

```diff
@@ -334,6 +334,119 @@ BOOST_AUTO_TEST_CASE(metadata_revert_strings)
 	BOOST_CHECK_EQUAL(metadata["settings"]["debug"]["revertStrings"], "strip");
 }
 
+BOOST_AUTO_TEST_CASE(metadata_license_missing)
+{
+	CompilerStack compilerStack;
+	char const* sourceCode = R"(
+		pragma solidity >=0.0;
+		contract C {
+		}
+	)";
+	compilerStack.setSources({{"", std::string(sourceCode)}});
+	BOOST_REQUIRE_MESSAGE(compilerStack.compile(), "Compiling contract failed");
+
+	std::string const& serialisedMetadata = compilerStack.metadata("C");
+	BOOST_CHECK(solidity::test::isValidMetadata(serialisedMetadata));
+	Json::Value metadata;
+	BOOST_REQUIRE(util::jsonParseStrict(serialisedMetadata, metadata));
+
+	BOOST_CHECK_EQUAL(metadata["sources"].size(), 1);
+	BOOST_CHECK(metadata["sources"].isMember(""));
+	BOOST_CHECK(!metadata["sources"][""].isMember("license"));
+}
+
+BOOST_AUTO_TEST_CASE(metadata_license_gpl3)
+{
+	CompilerStack compilerStack;
+	// Can't use a raw string here due to the stylechecker.
+	char const* sourceCode =
+		"// NOTE: we also add trailing whitespace after the license, to see it is trimmed."
+		"// SPDX-License-Identifier: GPL-3.0    "
+		"pragma solidity >=0.0;"
+		"contract C {"
+		"}";
+	compilerStack.setSources({{"", std::string(sourceCode)}});
+	BOOST_REQUIRE_MESSAGE(compilerStack.compile(), "Compiling contract failed");
+
+	std::string const& serialisedMetadata = compilerStack.metadata("C");
+	BOOST_CHECK(solidity::test::isValidMetadata(serialisedMetadata));
+	Json::Value metadata;
+	BOOST_REQUIRE(util::jsonParseStrict(serialisedMetadata, metadata));
+
+	BOOST_CHECK_EQUAL(metadata["sources"].size(), 1);
+	BOOST_CHECK(metadata["sources"].isMember(""));
+	BOOST_CHECK(metadata["sources"][""].isMember("license"));
+	BOOST_CHECK_EQUAL(metadata["sources"][""]["license"].asString(), "GPL-3.0");
```

Comment:
What do you think about putting the repeating part in a helper?
Or we could have a helper to validate the relevant part of the returned JSON (i.e. that the `sources` key is there, that there's only one source, etc.) and another that returns a list of licenses found in the metadata or `nullopt` so that we can do comparison here.

Reply:
Honestly I don't want to spend more time on this PR than neccesary. We should move standardjson/metadata tests into a similar system the other ones work. I think spending time on that compared to writing helpers in the existing boost tests may be better.

Reply:
Good point. It's fine then.